src/generic_core/social.ts

```diff
@@ -391,8 +401,7 @@ module Social {
         // TODO: Consider adding myself to the roster.
         if (client.clientId === this.myInstance.clientId &&
             client.status === UProxyClient.Status.OFFLINE) {
-          ui.showNotification('You have been logged out of ' + this.name);
-          core.logout({name: this.name, userId: this.myInstance.userId});
+          this.fulfillLogout_();
```

Comment:
I think you will want to call stopMonitor and also handleLogout on all users in the roster here as well (to close any socks-to-rtc or rtc-to-net connections).  Maybe we should setup a .then callback on onceLoggedOut_ to do that cleanup?

i.e. the "public logout" method below should call ```this.freedomApi_.logout().then(this.fulfillLogout.bind(this))``` and the then callback on lines 503-504 should include the stopMonitor() and handleLogout on all users that is curently in "public logout"